db/schema_tables.cc

```diff
@@ -1728,6 +1733,12 @@ mutation make_scylla_tables_mutation(schema_ptr table, api::timestamp_type times
         auto& cdef = *scylla_tables()->get_column_definition("partitioner");
         m.set_clustered_cell(ckey, cdef, atomic_cell::make_dead(timestamp, gc_clock::now()));
     }
+    // In-memory tables are deprecated since scylla-2024.1.0
+    if (auto in_memory_cdef = scylla_tables()->get_column_definition("in_memory")) {
+        // If IN_MEMORY_TABLES feature is enabled, just delete the unused column
+        // so we can safely drop the feature in the future, when it becomes unused by all live releases.
+        m.set_clustered_cell(ckey, *in_memory_cdef, atomic_cell::make_dead(timestamp, gc_clock::now()));
```

Comment:
We should do this only if the cluster in_memory_tables feature is enabled.  Or maybe just ignore this column on read and never write it.
The schema compaction strategy class InMemoryCompactionStrategy is enough, so we don't really need this boolean column in the schema table.